report2idea/dgadetection/dgadetection2idea.py

```diff
@@ -0,0 +1,71 @@
+#!/usr/bin/python3
+
+# In case we are in nemea/modules/report2idea/bruteforce and we want to import from repo:
+import os, sys
+sys.path.insert(0, os.path.join(os.path.dirname(__file__), "..", "..", "..", "nemea-framework", "pycommon"))
+
+import argparse
+
+# The whole functionality of reporting is here:
+from report2idea import *
+import socket
+
+
+# Moudle name, description and required input data format
+MODULE_NAME = "dgadetection"
+MODULE_DESC = "Converts output of DGA_detector module to IDEA."
+
+REQ_TYPE = pytrap.FMT_UNIREC
+REQ_FORMAT = "ipaddr SRC_IP,ipaddr DST_IP,time TIME_FIRST,time TIME_LAST,string DNS_Q_NAME"
+
+
+# Main conversion function
+def convert_to_idea(rec, opts=None):
+    """
+    Get fields from UniRec message 'rec' and convert it into an IDEA message (Python dict()).
+
+    rec - Record received on TRAP input interface (the report to convert).
+          Its format satisfies what was defined by REQ_TYPE and REQ_FORMAT.
+    opts - options parsed from command line (as returned by argparse.ArgumentParser)
+
+    Return report in IDEA format (as Python dict). If None is returned, the alert is skipped.
+    
+    """
+    idea = {
+        "Format": "IDEA0",
+        "ID": getRandomId(),
+        "DetectTime": getIDEAtime(rec.TIME_LAST),
+        #"CreateTime": getIDEAtime(),
+        "Category": [ "Intrusion.Botnet","test" ],
+        "Description": "Botnet trying to comunicate with control server using DGA",
+        "Source": [{
+            #"IP4": [rec.SRC_IP]
+         }],
+        "Target": [{    
+            #"IP4": [rec.DST_IP],
+            "Hostname": [rec.DNS_Q_NAME]
+        }],
```

Comment:
As Tomáš mentioned, both should be listed as Source. In IDEA, Source and Target doen't necessarily correspond with direction of communication, instead Source means the misbehaving party, Target is the victim. In case of bot-cc communication, both sides are "bad", so both should be listed as Source and distinguished by tags:
```
"Source": [{
    # Bot (address filled below)
    "Type": ["Botnet"]
},
{
    # CC server
    "Type": ["Botnet", "CC"],
    "Hostname": [rec.DNS_Q_NAME]
}],
```